### PR TITLE
feat: LSeq impl & Tests

### LINKED ISSUE #82: Implement built-in CRDTs: LSEQ (array type)
### Background
CRDT has its origin in the collaborative document use case. The document is modeled as an array of characters that can be concurrently edited (character insertion & deletion). There are lots of array/sequence CRDTs for this purpose. The two basic ones are LSEQ and RGA. This issue tracks LSEQ implementation and testing.

### Task
- implement 
- test
- document how this type works and its usage

### Resources
- https://hal.science/hal-00921633/document original paper
- https://www.bartoszsypytkowski.com/operation-based-crdts-arrays-1/
- https://docs.rs/crdts/latest/crdts/list/index.html the Rust crate for CRDT also implements their array type as LSEQ

### PR DESCRIPTION
**Pull request by joaopereira12**
resolves #82

### CODE REVIEW

**Comment by joaopereira12** [comment]
I'm marking this PR as draft because I just realized there's a function I want to change.

File: packages/crdt/src/builtins/LSeq/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+type VPointer = {
+    sequence: number[],
+    nodeId: string
+};
```
Comment (d-roak) [inline packages/crdt/src/builtins/LSeq/index.ts:4]:
```suggestion
type VPointer = {
    sequence: number,
    nodeId: string
};
```

Comment (joaopereira12) [reply packages/crdt/src/builtins/LSeq/index.ts:4]: Here the sequence needs to be a byte array. For example, to generate a key between [1] and [2] the result will be [1,1] (https://www.bartoszsypytkowski.com/operation-based-crdts-arrays-1/).

**Comment by d-roak** [comment]
not using crdt builtins anymore

**Pull request closed without merging**